src/wp-admin/post.php

```diff
@@ -239,26 +239,26 @@
 		check_admin_referer( 'trash-post_' . $post_id );
 
 		if ( ! $post ) {
-			wp_die( __( 'The item you are trying to move to the Trash no longer exists.' ) );
+			wp_die( __( 'The item you are trying to move to the Trash no longer exists.' ), 404 );
 		}
 
 		if ( ! $post_type_object ) {
-			wp_die( __( 'Invalid post type.' ) );
+			wp_die( __( 'Invalid post type.' ), 400 );
 		}
 
 		if ( ! current_user_can( 'delete_post', $post_id ) ) {
-			wp_die( __( 'Sorry, you are not allowed to move this item to the Trash.' ) );
+			wp_die( __( 'Sorry, you are not allowed to move this item to the Trash.' ), 403 );
 		}
 
 		$user_id = wp_check_post_lock( $post_id );
 		if ( $user_id ) {
 			$user = get_userdata( $user_id );
 			/* translators: %s: User's display name. */
-			wp_die( sprintf( __( 'You cannot move this item to the Trash. %s is currently editing.' ), $user->display_name ) );
+			wp_die( sprintf( __( 'You cannot move this item to the Trash. %s is currently editing.' ), $user->display_name ), 400 );
```

Comment:
Either the `409 Conflict` status can be used here as well, or the even more explicit `423 Locked` status: https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Status/423

```suggestion
			wp_die( sprintf( __( 'You cannot move this item to the Trash. %s is currently editing.' ), $user->display_name ), 423 );
```